pil-analyzer/src/pil_analyzer.rs

```diff
@@ -71,6 +75,7 @@ struct PILAnalyzer {
     symbol_counters: Option<Counters>,
     /// Symbols from the core that were added automatically but will not be printed.
     auto_added_symbols: HashSet<String>,
+    implementations: HashMap<String, Vec<(SourceRef, TraitImplementation<Expression>)>>,
```

Comment:
Please document.

Also maybe we could move the source ref into the implementation struct?